### PR TITLE
[Issue-381] docker-compose version

### PR DESCRIPTION
**Pull request by jpizagno**
## Motivation
This PR creates a docker-compose version of nrtSearch with 1 primary and 3 replicas.  This would not be a production setup, but would be helpful for 1) debugging and 2) demonstrating how to run nrtSearch.  It also includes an example of how to backup and restore the index from an S3 bucket.

## Implementation
I have added `docker-compose.yaml `file that starts the primary and 3 replicas.  There are config files, and a data file under `docker-compose-config`.  I added everything under `docker-compose-config` just to keep the main code base clean. There is documentation under the doc/ sub directory.

## Questions for Reviewers

1. Should I put the commands for the primary and/or replica [here](https://github.com/Yelp/nrtsearch/pull/417/files#diff-b3cb0c35cf5901a8c10950e35c67d1976733d3eb4592a0190f529fdcf424c5c4R48) in a  single script ?

## Testing
To test this PR, one can follow the commands in the `docs/docker_compose.rst` file.

### CODE REVIEW

File: docs/docker_compose.rst [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  shell$ docker -v
+  Docker version 20.10.8, build 3967b7d28e
+
+  shell$ docker-compose -v
+  docker-compose version 1.27.4, build 40524192
+
+1. Starting Containers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There is a Dockerfile in the main directory, which is used by docker-compose to build the image used for both the primary and replica nodes.   Currently there are 3 replicas started, but that can be increased in the docker-compose.yaml file.
```
[the remark this answers is not in the document]
Comment (jpizagno) [reply docs/docker_compose.rst:26]: thx, fixed.

File: docs/docker_compose.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+.. code-block::
+
+  shell% docker exec -it $PRIMARY_CONTAINER_ID sh
+  # ./build/install/nrtsearch/bin/lucene-client -h primary-node -p 8000 commit -i testIdx
+
+5. Replica: Search Should Work
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The search should now work on any of the replicas where the the index was started and registered on the primary.
```
[the remark this answers is not in the document]
Comment (jpizagno) [reply docs/docker_compose.rst:81]: Okay.  I see that now. I updated the documentation.  I renamed section 4 "Replication", and the code there just adds the documents.  

Note: I also fixed some grammatical mistakes or nits in other parts of the text. Feel free to review the entire document again.

File: docs/docker_compose.rst [hunk 3]
```diff
@@ -0,0 +1,128 @@
+Docker Compose
```
[the remark this answers is not in the document]
Comment (jpizagno) [reply docs/docker_compose.rst:1]: done. 

@sarthakn7 should I remove the references to "Docker Version" in introduction.rst?   
In my opinion, that is no longer necessary, and might just be too verbose (i.e. keep it clear and concise).

File: src/main/java/com/yelp/nrtsearch/LuceneServerModule.java [hunk 4]
```diff
@@ -20,6 +20,7 @@
 import com.amazonaws.auth.AnonymousAWSCredentials;
 import com.amazonaws.auth.profile.ProfileCredentialsProvider;
 import com.amazonaws.auth.profile.ProfilesConfigFile;
+import com.amazonaws.auth.profile.internal.BasicProfile;
```
[the remark this answers is not in the document]
Comment (jpizagno) [reply src/main/java/com/yelp/nrtsearch/LuceneServerModule.java]: done

File: docs/docker_compose.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpizagno) [reply docs/docker_compose.rst:1]: done. the introduction.rst should not  have any references to the "docker version"

File: docs/docker_compose.rst [hunk 5]
```diff
@@ -18,7 +18,36 @@ This has been tested with docker and docker-compose versions:
   Docker version 20.10.8, build 3967b7d28e
 
   shell$ docker-compose -v
-  docker-compose version 1.27.4, build 40524192
+  docker-compose version 1.27.4, build 40524192 # version 1.27 so the deploy.replicas option works
+
+0. PreSteps
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Start an AWS EC2 instance, with docker and git installed.  One needs docker-compose to run the example because EC2 does not come with docker-compose. One will also need a boto.cfg that can access a pre-existing S3 bucket.  This can be done like this (details may vary according to your AMI image):
```
Comment (sarthakn7) [inline docs/docker_compose.rst:26]: we can access S3 from a local machine as well right? Probably better to clarify Ec2 instance usage is not mandatory

**Review by sarthakn7 — APPROVED** [review]

**Pull request merged**